### PR TITLE
Add fatstack support, use reusable CI and update ragger

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

**Comment by xchapron-ledger** [comment]
@ericpassmore It's now fully ready! Please have a look :)

File: .github/workflows/build_and_functional_tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Build and run functional tests using ragger through reusable workflow
```
Comment (tdejoigny-ledger) [inline .github/workflows/build_and_functional_tests.yml:1]: Why all the CI worflows from boilerplate have not been included here (as below):
image

File: Makefile [hunk 2]
```diff
@@ -20,6 +20,10 @@ $(error Environment variable BOLOS_SDK is not set)
 endif
 include $(BOLOS_SDK)/Makefile.defines
 
+$(info TARGET_NAME=$(TARGET_NAME))
```
Comment (tdejoigny-ledger) [inline Makefile]: makefile not aligned with boilerplate makefile. this ligne is not present in Boilerplate makefile. And `APP_LOAD_PARAMS ` is not set here.

File: src/ui_nbgl.c [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    // valid page so return true
+    return true;
+}
+
+static void app_quit(void) {
+    // exit app here
+    os_sched_exit(-1);
+}
+
+static void controls_callback(int token, uint8_t state) {
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c:71]: should be interesting to have more comments for this function

File: src/ui_nbgl.c [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        if (state == 0 && is_data_allowed()) {
+            toogle_data_allowed();
+        } else if (state != 0 && !is_data_allowed()) {
+            toogle_data_allowed();
+        }
+    }
+}
+
+static void app_settings() {
+    nbgl_useCaseSettings(APPNAME, 0, 2, false, ui_idle, nav_callback, controls_callback);
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c]: usage  of `#define` helps to understand the code (for example here `0, 2, false`

File: src/ui_nbgl.c [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+static actionArgument_t bkp_args[MAX_TAG_VALUE_PAIRS_DISPLAYED];
+
+static char review_title[20];
+
+static void transaction_rejected(void) {
+    user_action_tx_cancel();
+}
+
+static void reject_confirmation(void) {
+    nbgl_useCaseConfirm("Reject transaction?", NULL, "Yes, Reject", "Go back to transaction", transaction_rejected);
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c:157]: perhaps a space between `transaction `and` ?` will increase the readability

File: src/ui_nbgl.c [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    pairList.callback = get_single_action_review_pair;
+    pairList.startIndex = 0;
+
+    nbgl_useCaseStaticReview(&pairList, &infoLongPress, "Reject transaction", review_choice);
+}
+
+
+void ui_display_single_action_sign_flow(void) {
+    if (txProcessingCtx.currentActionNumer > 1) {
+        snprintf(review_title, sizeof(review_title), "Review action #%d", txProcessingCtx.currentActionIndex);
```
Comment (tdejoigny-ledger) [inline src/ui_nbgl.c]: minor comment
`txProcessingCtx.currentActionIndex` is `uint32_t` type so
 in theory the string `static char review_title[20];` could have a size bigger than 20 char

**Review by tdejoigny-ledger — COMMENTED** [review]
In addition the `readme.md` file seems not to be up to date

File: .github/workflows/build_and_functional_tests.yml (hunk 1 above)
Comment (xchapron-ledger) [reply .github/workflows/build_and_functional_tests.yml:1]: As discussed, these jobs doesn't apply to this code base. At least not in its current state.

File: Makefile (hunk 2 above)
Comment (xchapron-ledger) [reply Makefile]: Check removed, APP_LOAD_PARAMS is well defined below.

File: src/ui_nbgl.c (hunk 3 above)
Comment (xchapron-ledger) [reply src/ui_nbgl.c:71]: added

File: src/ui_nbgl.c (hunk 4 above)
Comment (xchapron-ledger) [reply src/ui_nbgl.c]: done

File: src/ui_nbgl.c (hunk 5 above)
Comment (xchapron-ledger) [reply src/ui_nbgl.c:157]: No space before a `?` in english as far as I know.

File: src/ui_nbgl.c (hunk 6 above)
Comment (xchapron-ledger) [reply src/ui_nbgl.c]: I've sanitized this enforcing an `'\0'` at the end of the string.

**Pull request merged**